runtime/vm/jvminit.c

```diff
@@ -2902,59 +2916,59 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 
 		/* Since these two options are -XX, we don't need to consume them as they will be ignored */
 
-		if (strcmp(testString, VMOPT_XXNOSTACKTRACEINTHROWABLE) == 0) {
+		if (FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL) != -1) {
```

Comment:
There are two problems with these changes to use FIND_AND_CONSUME_ARG, the second of which is causing the extended test failure.
- this change is inefficient, this code is in a loop. It will check for these options every time through the loop, however FIND_AND_CONSUME_ARG only needs to be called once for each option
- the order of options is not taken into account by FIND_AND_CONSUME_ARG. For options that have both a + and - variation specified, we need to check which one comes last and use that one

Reply:
I have changed the structure for each if statement to something like the following:
`if ((FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL) != -1) || (FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXSTACKTRACEINTHROWABLE, NULL) != -1)) {
		if ((FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL) != -1) && (FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXSTACKTRACEINTHROWABLE, NULL) != -1)) {
			if (FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL) > FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXSTACKTRACEINTHROWABLE, NULL)) {
				vm->runtimeFlags |= J9_RUNTIME_OMIT_STACK_TRACES;
			} else {
				vm->runtimeFlags &= ~(UDATA)J9_RUNTIME_OMIT_STACK_TRACES;
			}
		} else {
			if(FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL) != -1) {
				vm->runtimeFlags |= J9_RUNTIME_OMIT_STACK_TRACES;
			} else if (FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXSTACKTRACEINTHROWABLE, NULL) != -1) {
				vm->runtimeFlags &= ~(UDATA)J9_RUNTIME_OMIT_STACK_TRACES;
			}
		}	
	}`

Reply:
On a second look, I found it much easier to do the following: 

![image](https://user-images.githubusercontent.com/33072072/53734581-67ba1700-3e52-11e9-9710-19c1131c3c8d.png)

Where I simply just add a check where I would check if both options exist first and consume the last one if both existed.

Reply:
Please find, and consume, each argument only once. Every time FIND_ is called there is a search through all the options.

Reply:
Okay, so I guess each statement would look like the following
`IDATA noStackTraceInThrowable = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOSTACKTRACEINTHROWABLE, NULL);
`IDATA StackTraceInThrowable = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXSTACKTRACEINTHROWABLE, NULL);
if ((noStackTraceInThrowable > stackTraceInThrowable) && (noStackTraceInThrowable != -1)) {
    vm->runtimeFlags |= J9_RUNTIME_OMIT_STACK_TRACES;
} else if ((stackTraceInThrowable > noStackTraceInThrowable) && (stackTraceInThrowable != -1)) {
    vm->runtimeFlags &= ~(UDATA)J9_RUNTIME_OMIT_STACK_TRACES;
}`

Reply:
Looking ahead, it seems like space is an issue if we declare an IDATA variable for each if statement. Perhaps we can have a temporary variable that can be reused for each if statement? If so, what should that variable be called?

Reply:
You don't need to check `<variable> != -1`. If it is greater than the alternate variable then it can't be -1.

Please use a different set of local variables for each option. Clarity and lack of bugs is more important that saving some stack space. If you use a different scope (i.e. `{` `}`) for checking each option, it will make it easier for the compiler to take care of space optimization.